src/sentry/static/sentry/app/views/stream.jsx

```diff
@@ -478,6 +523,37 @@ const Stream = createReactClass({
     });
   },
 
+  onLatestContextChange(context) {
+    if (this.state.hasEnvironmentsFeature) {
+      // Always query the currently active environment selection unless
+      // the environment parameter is part of the saved search
+      let environment = context.environment;
+
+      let query = this.state.query;
+
+      if (environment) {
+        query = objToQuery({
+          ...queryToObj(this.state.query),
+          environment: environment.name,
+        });
+      } else {
```

Comment:
again, maybe label these branches explicitly? i know the comment above is doing it, but labeled if branches help a lot when reading dense logic like this IMO